rclpy_components/test/test_component_manager.py

```diff
@@ -0,0 +1,198 @@
+        load_req.package_name = package_name
+        load_req.plugin_name = plugin_name
+
+        if node_name != "":
+            load_req.node_name = node_name
+        if node_namespace != "":
+            load_req.node_namespace = node_namespace
+
+        future = cls.load_cli.call_async(load_req)
+        rclpy.spin_until_future_complete(cls.helper_node, future)
+        return future.result()
+
+    @classmethod
+    def unload_node(cls, unique_id):
+        if not cls.unload_cli.wait_for_service(timeout_sec=5.0):
+            raise RuntimeError(f'No unload service found in /{cls.container_name}')
+
+        unload_req = UnloadNode.Request()
+        unload_req.unique_id = unique_id
+
+        future = cls.unload_cli.call_async(unload_req)
+        rclpy.spin_until_future_complete(cls.helper_node, future)
+        return future.result()
+
+    @classmethod
+    def list_nodes(cls):
+        if not cls.list_cli.wait_for_service(timeout_sec=5.0):
+            raise RuntimeError(f'No list service found in {cls.container_name}')
+        list_req = ListNodes.Request()
+        future = cls.list_cli.call_async(list_req)
+        rclpy.spin_until_future_complete(cls.helper_node, future)
+        return future.result()
+
+    def load_node_test(self):
+        load_res = self.__class__.load_node(TEST_COMPOSITION, TEST_COMPOSITION_FOO)
+        assert load_res.success is True
+        assert load_res.error_message == ""
+        assert load_res.unique_id == 1
+        assert load_res.full_node_name == TEST_COMPOSITION_NODE_NAME
+
+        node_name = "renamed_node"
+        node_ns = 'testing_ns'
+        remap_load_res = self.__class__.load_node(
+            TEST_COMPOSITION, TEST_COMPOSITION_FOO, node_name=node_name,
+            node_namespace=node_ns)
+        assert remap_load_res.success is True
+        assert remap_load_res.error_message == ""
+        assert remap_load_res.unique_id == 2
+        assert remap_load_res.full_node_name == f'/{node_ns}/{node_name}'
+
+        list_res: ListNodes.Response = self.__class__.list_nodes()
+        assert len(list_res.unique_ids) == len(list_res.full_node_names) == 2
+
+    def unload_node_test(self):
+        if not self.__class__.unload_cli.wait_for_service(timeout_sec=5.0):
+            raise RuntimeError(f'no unload service found in {self.__class__.container_name}')
+
+        unload_res: UnloadNode.Response = self.__class__.unload_node(1)
+        assert unload_res.success is True
+        assert unload_res.error_message == ""
+
+        # Should be only 1 node left
+        list_res: ListNodes.Response = self.__class__.list_nodes()
+        assert len(list_res.full_node_names) == len(list_res.unique_ids) == 1
+
+        # The index definitely won't exist
+        unload_res: UnloadNode.Response = self.__class__.unload_node(1000)
+        assert unload_res.error_message != ""
+        assert unload_res.success is False
+        list_res: ListNodes.Response = self.__class__.list_nodes()
```

Comment:
nit - don't need to specify `__class__` when calling class methods;  `self.list_nodes()` works